src/transformers/models/fastspeech2/configuration_fastspeech2.py

```diff
@@ -0,0 +1,154 @@
+    # See all FastSpeech2 models at https://huggingface.co/models?filter=fastspeech2
+}
+
+
+class FastSpeech2Config(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`FastSpeech2Model`]. It is used to instantiate an
+    FastSpeech2 model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the FastSpeech2
+    [fastspeech2](https://huggingface.co/fastspeech2) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        vocab_size (`int`, *optional*, defaults to 75):
+            Vocabulary size of the FastSpeech2 model. Defines the number of different tokens that can be represented by
+            the `inputs_ids` passed when calling [`~FastSpeech2Model`] or [`~TFFastSpeech2Model`].
+        encoder_layers (`int`, *optional*, defaults to 4):
+            Number of hidden layers in the FastSpeech2 encoder.
+        decoder_layers (`int`, *optional*, defaults to 4):
+            Number of hidden layers in the FastSpeech2 decoder.
+        encoder_attention_heads (`int`, *optional*, defaults to 2):
+            Number of attention heads for each attention layer in the FastSpeech2 encoder.
+        decoder_attention_heads (`int`, *optional*, defaults to 2):
+            Number of attention heads for each attention layer in the FastSpeech2 decoder.
+        intermediate_size (`int`, *optional*, defaults to 3072):
+            Dimension of the "intermediate" (i.e., feed-forward) layer in the Transformer encoder.
+        hidden_dropout_prob (`float`, *optional*, defaults to 0.1):
+            The dropout probabilitiy for all fully connected layers in the embeddings, encoder, and pooler.
+        attention_probs_dropout_prob (`float`, *optional*, defaults to 0.1):
+            The dropout ratio for the attention probabilities.
+        max_source_positions (`int`, *optional*, defaults to 1024):
+            The maximum sequence length that this model might ever be used with. Typically set this to something large
+            just in case (e.g., 512 or 1024 or 2048).
+        initializer_range (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        Example:
+
+    ```python
+    >>> from transformers import FastSpeech2Model, FastSpeech2Config
+
+    >>> # Initializing a FastSpeech2 fastspeech2 style configuration
+    >>> configuration = FastSpeech2Config()
+
+    >>> # Initializing a model from the fastspeech2 style configuration
+    >>> model = FastSpeech2Model(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+    model_type = "fastspeech2"
```

Comment:
@anton-l - do you think it's ok if the "general" `fastspeech2` name is taken here even thought the model is from MS originally? Fine for me personally, if Facebook's fastspeech2 is the "best" OS fastspeech implementation

Should we ask the MS authors if they are ok with this just to be sure it's fine?

Reply:
Although I do think `fairseq`'s FastSpeech2 is likely the "best" OS implementation out there, I'm not too confident about the naming. I don't think we can ever put it under `microsoft/fastspeech2`; if anything, it should be `facebook/fastspeech2`, or maybe even `facebook/fairseq-fastspeech2`.

Has there ever been a case where we ported an "unofficial" model to the transformers library (although `fairseq` is a pretty reputable source)?

Reply:
Think it's a first ;-) But fine to leave it as is here actually